pkg/services/govmomi/service.go

```diff
@@ -159,7 +162,7 @@ func (vms *VMService) DestroyVM(ctx *context.VMContext) (infrav1.VirtualMachine,
 	if err != nil {
 		// If the VM's MoRef could not be found then the VM no longer exists. This
 		// is the desired state.
-		if isNotFound(err) || isFolderNotFound(err) {
+		if errors.As(err,find.NotFoundError{}) || errors.As(err,find.NotFoundError{}) {
```

Comment:
Should this be `if errors.As(err, errNotFound{}) || errors.As(err,find.NotFoundError{}) {` ?